### PR TITLE
feat: implement make with requirements.txt

### LINKED ISSUE #2: Set up Python, FastAPI for backend server
# Problem
Installing Python using the FastAPI framework. Creating a requirements.txt file and a makefile.

Read the FastAPI documentation:
https://fastapi.tiangolo.com/tutorial/

# Success
Running the makefile commands on Windows and Mac.

### Requirements

- [ ] Create new branch from dev (name/issue Ex: yurika/setup-python)
- [ ] Install Python
- [ ] Install FastAPI
- [ ] Create a makefile with the following commands (should work with Windows and Mac)
  - [ ] Create virtual environment
  - [ ] Installing dependencies
  - [ ] Running the server
- [ ] Create a requirements.txt
- [ ] Update backend repository README to include python installation documentation under Prerequisites, and update Install Dependencies section

### PR DESCRIPTION
**Pull request by NabbeunNabi**
# Description
I installed a `Makefile` which has commands to run what developers need to run the files we are using right now. I would like someone on Mac to make sure the commands work on Mac and that they can run the `Makefile` commands as I tested them on Windows

# Closes issue(s)

Resolves #2 

# How to test
To test you want to go to my branch and try to run the following commands in this order

1.  `make venv` : This should install the venv on your local computer
2.  `make install`: This should install the requirements.txt
3.  `make run`: This should run the endpoint for `/` and you should see `Hello World`
4.  `make clean`: At the moment doesn't do anything noticeable
5.  `make freeze`: Should update the requirements.txt file if you add or change dependencies. Though at the moment for testing it won't change any as you aren't adding any

**THIS NEEDS TO BE TESTED BY SOMEONE ON A MAC TO MAKE SURE THE COMMAND `MAKE` IS WORKING PROPERLY ON THAT OS. I CODE ON A WINDOWS**

# Changes include
FastApi was added as a dependency. 

# Checklist

- [x] I have tested this code
- [x] I have updated the README

### CODE REVIEW

**Review by Anissa3005 — CHANGES_REQUESTED** [review]
Looks great ✨working with a make file for the first time can be bit challenging since it's less straightforward. 

- Some things we need to consider is, which version of python are we using?  We need to make sure that everyone is using the same version. So we need to implement that somewhere.
- In the `make` file, if I just would run a command without activated my venv that can cause trouble. So we need to think about how we can avoid that. Since when I did `make run` it didn't activated for me. So we can just mention it in the readme or we can add it in the make file. (we can pair on it if you need help)

File: README.md [hunk 1]
```diff
@@ -4,11 +4,25 @@
 # Setup
 
 ## 1. Install dependencies
+ ### On Windows
```
[the remark this answers is not in the document]
Comment (Anissa3005) [reply README.md:7]: Maybe we can also consider adding recommendations for VSCode extensions in our readme. Like extensions for Python, makefile,...

[the remark this answers is not in the document]
Comment (NabbeunNabi) [reply README.md]: I did update the Readme a bit

[the remark this answers is not in the document]
Comment (hccleveland) [reply README.md]: ✅ This also works on Mac. Just the venv activation is outstanding.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# Define default target
+.DEFAULT_GOAL := help
+
+# Display help message
+help:
+	@echo "Please use 'make <target>' where <target> is one of:"
+	@echo "  venv        to create a virtual environment"
+	@echo "  install     to install dependencies"
+	@echo "  run         to run the FastAPI server"
```
Comment (hccleveland) [inline Makefile:47]: Should also change to `runserver`.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	@echo "  clean       to clean up the virtual environment"
```
Comment (hccleveland) [inline Makefile:48]: Do we also want to add `freeze` to the list?

File: README.md [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
+2.  Run `make venv`
+3.  Run `make install`
+
+### On Mac
+
+1.  Run `make venv`
+2.  Start your `venv`.
+    - On Windows you can start the `venv` by running `source venv/Scripts/activate`
+    - On Mac and Linux you can start the `venv` by running `source venv/bin/activate`
+3.  Run `make install`
```
Comment (hccleveland) [inline README.md:27]: Do we now want to specify what requires manual activation of the virtual environment and what is activated automatically?

**Review by hccleveland — CHANGES_REQUESTED** [review]
Just a few points of cleaning up. Sorry to miss these earlier.

**Review by hccleveland — APPROVED** [review]
LGTM. Tested locally on Mac.

**Review by Anissa3005 — APPROVED** [review]
LGTM

**Pull request merged**